askama_derive/src/generator.rs

```diff
@@ -987,6 +989,9 @@ impl<'a> Generator<'a> {
         if block_fragment_write {
             self.buf_writable.discard = true;
         }
+        if buf.discard != prev_buf_discard {
```

Comment:
This seems surprising to me -- why do we only want to write if `buf.discard` is not equal to `prev_buf_discard` vs writing it out if `buf.discard` is just `false`?

Reply:
Interesting question. In my previous PR about this, if I wrote too early, we had content from sub-block that was rendered before its parent's, which was problematic. So in this case, if `buf.discard != prev_buf_discard`, then it's like that the parent is actually not something being displayed and so we should flush everything before the `discard` is `true` again.

Does my reasoning sound good to you and do you see potential corner-cases?

Reply:
Just thought: do you want me to add a code comment to explain this?

Reply:
Yes, that would be a good improvement.

Reply:
Done!

Reply:
This comment is too long IMO and still doesn't make it very clear what case this is trying to fix/prevent. I want a comment here in like 4-5 lines that explains conceptually what case it is trying to handle.

Reply:
Updated the comment.